packages/zksync-dai-bridge/contracts/l2/dai.sol

```diff
@@ -0,0 +1,234 @@
+          chainId,
+          address(this)
+        )
+      );
+  }
+
+  function DOMAIN_SEPARATOR() external view returns (bytes32) {
+    uint256 chainId;
+    assembly {
+      chainId := chainid()
+    }
+    return chainId == deploymentChainId ? _DOMAIN_SEPARATOR : _calculateDomainSeparator(chainId);
+  }
+
+  // --- ERC20 Mutations ---
+  function transfer(address to, uint256 value) external returns (bool) {
+    require(to != address(0) && to != address(this), "Dai/invalid-address");
+    uint256 balance = balanceOf[msg.sender];
+    require(balance >= value, "Dai/insufficient-balance");
+
+    balanceOf[msg.sender] = balance - value;
+    balanceOf[to] += value;
+
+    emit Transfer(msg.sender, to, value);
+
+    return true;
+  }
+
+  function transferFrom(
+    address from,
+    address to,
+    uint256 value
+  ) external returns (bool) {
+    require(to != address(0) && to != address(this), "Dai/invalid-address");
+    uint256 balance = balanceOf[from];
+    require(balance >= value, "Dai/insufficient-balance");
+
+    if (from != msg.sender) {
+      uint256 allowed = allowance[from][msg.sender];
+      if (allowed != type(uint256).max) {
+        require(allowed >= value, "Dai/insufficient-allowance");
+
+        allowance[from][msg.sender] = allowed - value;
+      }
+    }
+
+    balanceOf[from] = balance - value;
+    balanceOf[to] += value;
+
+    emit Transfer(from, to, value);
+
+    return true;
+  }
+
+  function approve(address spender, uint256 value) external returns (bool) {
+    allowance[msg.sender][spender] = value;
+
+    emit Approval(msg.sender, spender, value);
+
+    return true;
+  }
+
+  function increaseAllowance(address spender, uint256 addedValue) external returns (bool) {
+    uint256 newValue = _add(allowance[msg.sender][spender], addedValue);
+    allowance[msg.sender][spender] = newValue;
+
+    emit Approval(msg.sender, spender, newValue);
+
+    return true;
+  }
+
+  function decreaseAllowance(address spender, uint256 subtractedValue) external returns (bool) {
+    uint256 allowed = allowance[msg.sender][spender];
+    require(allowed >= subtractedValue, "Dai/insufficient-allowance");
+    allowed = allowed - subtractedValue;
+    allowance[msg.sender][spender] = allowed;
+
+    emit Approval(msg.sender, spender, allowed);
+
+    return true;
+  }
+
+  // --- Mint/Burn ---
+  function mint(address to, uint256 value) external auth {
+    require(to != address(0) && to != address(this), "Dai/invalid-address");
+    balanceOf[to] = balanceOf[to] + value; // note: we don't need an overflow check here b/c balanceOf[to] <= totalSupply and there is an overflow check below
+    totalSupply = _add(totalSupply, value);
+
+    emit Transfer(address(0), to, value);
+  }
+
+  function burn(address from, uint256 value) external {
+    uint256 balance = balanceOf[from];
+    require(balance >= value, "Dai/insufficient-balance");
+
+    if (from != msg.sender && wards[msg.sender] != 1) {
```

Comment:
Remove wards check.

Reply:
Removed in #[289b411](https://github.com/makerdao/xdomain/pull/72/commits/289b41198d3cbf422dec49ce40d580fa09af8979)